### PR TITLE
aep extension bug fixes (commerce ML)

### PR DESCRIPTION
**Pull request by aniham**
- upgrade to latest version of collector
- generate search results view event
- remove context from search events (should be picked up from data layer)
- return facets as list instead of null (schema validation failures)

### CODE REVIEW

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/storefrontcontext/SearchResultsStorefrontContextImpl.java [hunk 1]
```diff
@@ -80,7 +81,7 @@ public int getPerPage() {
     @Override
     public List<SearchFacet> getFacets() {
         // TODO: return the facets as we have them available in our result set
-        return null;
+        return new ArrayList<SearchFacet>();
```
Comment (Buuhuu) [inline bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/storefrontcontext/SearchResultsStorefrontContextImpl.java:84]: Why is that needed? This is a change of behaviour in the API. Imaging, customer code checks the return value for `null`. That would not work anymore after that change.

Comment (aniham) [reply bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/storefrontcontext/SearchResultsStorefrontContextImpl.java:84]: @Buuhuu this method is only referenced when setting StorefrontContext. I don't believe any custom code could or would be using it. Moreover, as I mentioned in the PR summary, this now breaks how events are received for data collection. So. It looks like it was originally implemented wrong and never caught as a bug in testing.

**Review by Buuhuu — APPROVED** [review]

**Comment by herzog31** [comment]
Replaced by #991
